core/trino-main/src/main/java/io/trino/execution/querystats/QueryRuleStats.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.querystats;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.stats.TDigest;
+import io.trino.plugin.base.metrics.TDigestHistogram;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class QueryRuleStats
```

Comment:
We already have `io.trino.sql.planner.iterative.RuleStats`. Why do we need another one?

Reply:
`RuleStats` is not JSON serializable and it is used in a different context.

Reply:
One `RuleStats` class removed.

Reply:
To clarify, class `RuleInvocationStats` was removed. Both classes (`RuleStats` and `QueryPlanOptimizerStats`) are used in different context, one is used for JMX reporting and second one to expose statistics for `QueryInfo` JSON for `Rule` and `PlanOptimizer`.